[18.0] base_multi_company : Found unexpected attributes on multi.company.abstract.tester: company_id company_ids

## Module
base_multi_company

## Describe the bug
Since odoo/odoo@de056cc784a3bbe2575fd3c9e81ca62e73c362d4 a new checker for methods being set on models during tests has been introduced
As `base_multi_company` is using a fake class in tests, the following error is raised for the 8 tests of the module :
```
2026-02-20 04:09:30,013 329 ERROR odoo odoo.addons.base_multi_company.tests.test_multi_company_abstract: FAIL: TestMultiCompanyAbstract.test_set_company_id
2026-02-20T04:10:05.2288507Z Traceback (most recent call last):
2026-02-20T04:10:05.2288839Z   File "/opt/odoo-venv/bin/coverage", line 6, in <module>
2026-02-20T04:10:05.2289105Z     sys.exit(main())
2026-02-20T04:10:05.2289443Z   File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 1163, in main
2026-02-20T04:10:05.2289864Z     status = CoverageScript().command_line(argv)
2026-02-20T04:10:05.2290297Z   File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 853, in command_line
2026-02-20T04:10:05.2290733Z     return self.do_run(options, args)
2026-02-20T04:10:05.2291110Z   File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/cmdline.py", line 1042, in do_run
2026-02-20T04:10:05.2291498Z     runner.run()
2026-02-20T04:10:05.2291970Z   File "/opt/odoo-venv/lib/python3.10/site-packages/coverage/execfile.py", line 213, in run
2026-02-20T04:10:05.2292373Z     exec(code, main_mod.__dict__)
2026-02-20T04:10:05.2292622Z   File "/opt/odoo-venv/bin/odoo", line 8, in <module>
2026-02-20T04:10:05.2292878Z     odoo.cli.main()
2026-02-20T04:10:05.2293101Z   File "/opt/odoo/odoo/cli/command.py", line 76, in main
2026-02-20T04:10:05.2293354Z     o.run(args)
2026-02-20T04:10:05.2293560Z   File "/opt/odoo/odoo/cli/server.py", line 182, in run
2026-02-20T04:10:05.2293809Z     main(args)
2026-02-20T04:10:05.2294017Z   File "/opt/odoo/odoo/cli/server.py", line 175, in main
2026-02-20T04:10:05.2294353Z     rc = odoo.service.server.start(preload=preload, stop=stop)
2026-02-20T04:10:05.2294801Z   File "/opt/odoo/odoo/service/server.py", line 1462, in start
2026-02-20T04:10:05.2295101Z     rc = server.run(preload, stop)
2026-02-20T04:10:05.2295370Z   File "/opt/odoo/odoo/service/server.py", line 627, in run
2026-02-20T04:10:05.2295659Z     rc = preload_registries(preload)
2026-02-20T04:10:05.2295984Z   File "/opt/odoo/odoo/service/server.py", line 1366, in preload_registries
2026-02-20T04:10:05.2296395Z     registry = Registry.new(dbname, update_module=update_module)
2026-02-20T04:10:05.2296713Z   File "<decorator-gen-13>", line 2, in new
2026-02-20T04:10:05.2297001Z   File "/opt/odoo/odoo/tools/func.py", line 97, in locked
2026-02-20T04:10:05.2297273Z     return func(inst, *args, **kwargs)
2026-02-20T04:10:05.2297557Z   File "/opt/odoo/odoo/modules/registry.py", line 129, in new
2026-02-20T04:10:05.2297942Z     odoo.modules.load_modules(registry, force_demo, status, update_module)
2026-02-20T04:10:05.2298348Z   File "/opt/odoo/odoo/modules/loading.py", line 489, in load_modules
2026-02-20T04:10:05.2298769Z     processed_modules += load_marked_modules(env, graph,
2026-02-20T04:10:05.2299140Z   File "/opt/odoo/odoo/modules/loading.py", line 365, in load_marked_modules
2026-02-20T04:10:05.2299493Z     loaded, processed = load_module_graph(
2026-02-20T04:10:05.2299820Z   File "/opt/odoo/odoo/modules/loading.py", line 284, in load_module_graph
2026-02-20T04:10:05.2300214Z     test_results = loader.run_suite(suite, global_report=report)
2026-02-20T04:10:05.2300572Z   File "/opt/odoo/odoo/tests/loader.py", line 118, in run_suite
2026-02-20T04:10:05.2300936Z     suite(results)
2026-02-20T04:10:05.2301321Z AssertionError: Found unexpected attributes on multi.company.abstract.tester: company_ids company_id
```
This was encountered in #954 and #955 


## To Reproduce
**Affected versions**: 18.0

Steps to reproduce the behavior:
1. Just run test suite in any module of the repository

--- a/base_multi_company/tests/test_multi_company_abstract.py
+++ b/base_multi_company/tests/test_multi_company_abstract.py
@@ -9,28 +9,27 @@ from odoo.tests import common
 
 
 class TestMultiCompanyAbstract(common.TransactionCase):
-    @classmethod
-    def setUpClass(cls):
-        super().setUpClass()
-        cls.loader = FakeModelLoader(cls.env, cls.__module__)
-        cls.loader.backup_registry()
+    def setUp(self):
+        super().setUp()
+        self.loader = FakeModelLoader(self.env, self.__module__)
+        self.loader.backup_registry()
 
         # The fake class is imported here !! After the backup_registry
         from .multi_company_abstract_tester import MultiCompanyAbstractTester
 
-        cls.loader.update_registry((MultiCompanyAbstractTester,))
+        self.loader.update_registry((MultiCompanyAbstractTester,))
 
-        cls.test_model = cls.env["multi.company.abstract.tester"]
+        self.test_model = self.env["multi.company.abstract.tester"]
 
-        cls.tester_model = cls.env["ir.model"].search(
+        self.tester_model = self.env["ir.model"].search(
             [("model", "=", "multi.company.abstract.tester")]
         )
 
         # Access record:
-        cls.env["ir.model.access"].create(
+        self.env["ir.model.access"].create(
             {
                 "name": "access.tester",
-                "model_id": cls.tester_model.id,
+                "model_id": self.tester_model.id,
                 "perm_read": 1,
                 "perm_write": 1,
                 "perm_create": 1,
@@ -38,16 +37,15 @@ class TestMultiCompanyAbstract(common.TransactionCase):
             }
         )
 
-        cls.record_1 = cls.test_model.create({"name": "test"})
-        cls.company_1 = cls.env.company
-        cls.company_2 = cls.env["res.company"].create(
+        self.record_1 = self.test_model.create({"name": "test"})
+        self.company_1 = self.env.company
+        self.company_2 = self.env["res.company"].create(
             {"name": "Test Co 2", "email": "base_multi_company@test.com"}
         )
 
-    @classmethod
-    def tearDownClass(cls):
-        cls.loader.restore_registry()
-        return super().tearDownClass()
+    def tearDown(self):
+        self.loader.restore_registry()
+        return super().tearDown()
 
     def add_company(self, company):
         """Add company to the test record."""

